scipy/stats/tests/test_survival.py

```diff
@@ -0,0 +1,51 @@
+import pytest
+import numpy as np
+from numpy.testing import assert_equal
+from scipy import stats
+
+class TestSurvival:
+    def test_input_validation(self):
+        message = '`sample` must be a one-dimensional sequence.'
+        with pytest.raises(ValueError, match=message):
+            stats.ecdf([[1]])
+        with pytest.raises(ValueError, match=message):
+            stats.ecdf(1)
```

Comment:
Some folks might be tempted to pass n-D samples. I think we should clarify in the doc and also specifically test that.

Looking at the error message, it's also not clear that we want more than 1 obs.